src/account.ts

```diff
@@ -31,10 +31,16 @@ export const isZeroAddress = function(address: string): boolean {
 
 /**
  * Returns a checksummed address.
+ *
+ * If a chainId is provided, the result will be an EIP-1191 checksum. Otherwise, it will be an
+ * EIP-55 checksum.
  */
-export const toChecksumAddress = function(address: string): string {
+export const toChecksumAddress = function(address: string, chainId?: number): string {
   address = ethjsUtil.stripHexPrefix(address).toLowerCase()
-  const hash = keccak(address).toString('hex')
+
+  const prefix = chainId !== undefined ? chainId.toString() + '0x' : ''
```

Comment:
The documentation makes it clear, but if someone seeing `chainId` as the second parameter provides an Ethereum chainId (e.g. `1`) then the checksum will fail.

One alternative is to provide a second function `toEIP1991ChecksumAddress` (maybe with a better name :smile:).

I don't have a strong opinon however. @holgerd77 what do you think?

Reply:
That's a good point.

I made it this way to keep it similar to other implementations, like [web3.js' one](https://github.com/ethereum/web3.js/blob/aaf26c8806bc9fb60cf6dcb6658104963c6c7fc7/packages/web3-utils/src/index.js#L261)

Another alternative is to rename the second param to something like `eip1191ChainId`

Reply:
I would have a tendency to the renamed parameter. @alcuadrado can you also add "(prefixed with chainId)" and "(no prefix)" or some similar text after the respective EIP mentions in the JSDocs? Think this will also make things more clear, I often find these plain EIP mentions in our docs not very helpful if one is not totally in the EIP specs.

Reply:
One can also add some "Be are that checksums will be different" or so sentence.

Reply:
Are -> aware

Reply:
I changed the param name and updated the doc.